packages/compass-e2e-tests/tests/search-indexes.test.ts

```diff
@@ -0,0 +1,200 @@
+      });
+      afterEach(async function () {
+        await afterEachTestRun(this.currentTest);
+      });
+      it('allows users to create a regular index', async function () {
+        const indexName = await browser.createIndex({
+          fieldName: 'e2e_tests_index',
+          indexType: 'text',
+        });
+        await browser.dropIndex(indexName);
+      });
+      it('renders search indexes tab disabled', async function () {
+        const searchTab = await browser.$(
+          Selectors.indexesSegmentedTab('search-indexes')
+        );
+        const isTabClickable = await searchTab.isClickable();
+        expect(isTabClickable).to.be.false;
+      });
+    });
+  }
+
+  for (const { name, connectionString } of connectionsWithSearchSupport) {
+    context(`supports search indexes in ${name}`, function () {
+      const INDEX_NAME = 'e2e_search_index';
+      const INDEX_DEFINITION = JSON.stringify({
+        mappings: {
+          dynamic: true,
+        },
+      });
+
+      before(function () {
+        if (!connectionString) {
+          return this.skip();
+        }
+      });
+      beforeEach(async function () {
+        await beforeEachTestRun(connectionString!);
+
+        // Try to drop the existing search indexes. As we started afresh,
+        // the collection should not have any search index. However this
+        // is to just avoid any race condition.
+        {
+          const searchIndexes = await dbInstance
+            .collection(COLL_NAME)
+            .listSearchIndexes()
+            .toArray();
+
+          if (searchIndexes.length > 0) {
+            await Promise.all(
+              searchIndexes.map(({ name }) =>
+                dbInstance.collection(COLL_NAME).dropSearchIndex(name)
+              )
+            );
+            // Wait for a minute to ensure the indexes have been deleted.
+            await wait(60000);
+          }
+        }
+      });
+      afterEach(async function () {
+        await afterEachTestRun(this.currentTest);
+      });
+      it('allows users to create a regular indexes', async function () {
+        await browser.clickVisible(
+          Selectors.indexesSegmentedTab('regular-indexes')
+        );
+        const indexName = await browser.createIndex({
+          fieldName: 'e2e_tests_index',
+          indexType: 'text',
+        });
+        await browser.dropIndex(indexName);
+      });
+      it('allows users to create, view and drop search indexes', async function () {
+        await browser.clickVisible(
+          Selectors.indexesSegmentedTab('search-indexes')
+        );
+        await browser.createSearchIndex(INDEX_NAME, INDEX_DEFINITION);
+        await browser.waitForAnimations(Selectors.SearchIndexList);
+
+        // View it
+        await browser
+          .$(Selectors.searchIndexRow(INDEX_NAME))
+          .waitForDisplayed();
+
+        // Drop it
+        await browser.dropSearchIndex(INDEX_NAME);
```

Comment:
Maybe a good idea to wait and see if the index eventually goes away from the list too. Or is that a TODO for once we poll?

Reply:
i added an expectation to check if the status changes to `DELETING` in [2aab6d3](https://github.com/mongodb-js/compass/pull/4832/commits/2aab6d34c88449de8b95e2f3dca652abe7cb81c1)